src/averageAges.js

```diff
@@ -55,7 +79,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothersList = people.map(human => human.mother);
+
+  const mothersWithChildrens = people.filter(human =>
+    mothersList.includes(human.name)
+  );
+
+  const mothersNames = mothersWithChildrens.map(human => human.name);
+
+  const kidsHasMothers = !onlyWithSon
+    ? people.filter(human => mothersNames.includes(human.mother))
+    : people.filter(human => mothersNames.includes(human.mother)
+      && human.sex === 'm');
+
+  const mothers = {};
+
+  mothersWithChildrens.map(function(human) {
+    mothers[human.name] = human.born;
+  });
```

Comment:
You can make it simpler
Find all children using filter and then by map convert it to array of age differences